.github/workflows/shared-run-e2e-tests.yaml

```diff
@@ -0,0 +1,141 @@
+      core_image_version:
+        description: 'The version of UID2 core image'
+        type: string
+        default: '2.15.8-d2e5469826-default'
+      optout_image_version:
+        description: 'The version of UID2 optout image'
+        type: string
+        default: '2.7.3-4f813a8ad9-default'
+      e2e_image_version:
+        description: 'The version of E2E image'
+        type: string
+        default: 'latest'
+      operator_type:
+        description: 'The type of operator [either public or private]'
+        type: string
+        default: 'public'
+      uid2_e2e_identity_scope:
+        description: 'Environment variable to run the E2E test'
+        required: false
+        type: string
+        default: 'UID2'
+      uid2_e2e_pipeline_operator_type:
+        description: 'Environment variable to run the E2E test'
+        required: false
+        type: string
+        default: 'PUBLIC'
+      uid2_e2e_pipeline_operator_url:
+        description: 'Environment variable to run the E2E test'
+        required: false
+        type: string
+        default: 'http://publicoperator:8080'
+      uid2_e2e_pipeline_operator_cloud_provider:
+        description: 'Environment variable to run the E2E test'
+        required: false
+        type: string
+        default: 'PUBLIC'
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  e2e-test:
+    name: E2E Test
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: read
+      id-token: write
+    steps:
+      - name: Log in to the Docker container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          # we use personal access token here since we need to pull images from private repo (core)
+          password: ${{ secrets.GHCR_PAT }}
+
+      - name: Checkout full history
+        uses: actions/checkout@v3
+
+      - name: Checkout uid2-core repo
+        uses: actions/checkout@v3
+        with:
+          repository: IABTechLab/uid2-core
+          token: ${{ secrets.GHCR_PAT }}
+          path: uid2-core
+
+      - name: Checkout uid2-optout repo
+        uses: actions/checkout@v3
+        with:
+          repository: IABTechLab/uid2-optout
+          token: ${{ secrets.GHCR_PAT }}
+          path: uid2-optout
+
+      - name: Checkout uid2-admin repo
+        uses: actions/checkout@v3
+        with:
+          repository: IABTechLab/uid2-admin
+          token: ${{ secrets.GHCR_PAT }}
+          path: uid2-admin
+
+      - name: Checkout uid2-shared-actions repo
+        uses: actions/checkout@v3
+        with:
+          ref: kcc-UID2-2338-build-test-pipeline-for-operator
+          repository: IABTechLab/uid2-shared-actions
+          token: ${{ secrets.GHCR_PAT }}
+          path: uid2-shared-actions
+
+      - name: Checkout uid2-e2e repo
+        uses: actions/checkout@v3
+        with:
+          ref: kcc-UID2-2338-build-test-pipeline-for-operator
```

Comment:
This will be removed once the pr in e2e folder is merged: https://github.com/IABTechLab/uid2-e2e/pull/4